flowermd/tests/base/test_simulation.py

```diff
@@ -162,48 +172,20 @@ def test_update_volume_density(self, benzene_system):
     def test_update_volume_by_density_factor(self, benzene_system):
         sim = Simulation.from_system(benzene_system)
         init_density = copy.deepcopy(sim.density)
+        target_box = get_target_box_mass_density(
+            density=init_density * 5, mass=sim.mass.to(u.g)
+        )
         sim.run_update_volume(
             kT=1.0,
             tau_kt=0.01,
             n_steps=500,
             period=1,
-            final_density=sim.density * 5,
+            final_box_lengths=target_box,
         )
         assert np.isclose(
             sim.density.value, (init_density * 5).value, atol=1e-4
         )
 
-    def test_update_volume_missing_values(self, benzene_system):
-        sim = Simulation.from_system(benzene_system)
-        with pytest.raises(ValueError):
-            sim.run_update_volume(kT=1.0, tau_kt=0.01, n_steps=500, period=1)
-
-    def test_update_volume_two_values(self, benzene_system):
-        sim = Simulation.from_system(benzene_system)
-        with pytest.raises(ValueError):
-            sim.run_update_volume(
-                kT=1.0,
-                tau_kt=0.01,
-                n_steps=500,
-                period=1,
-                final_box_lengths=sim.box_lengths_reduced * 0.5,
-                final_density=0.1,
-            )
-
-    # def test_update_volume_with_density_no_ref_values(self, benzene_system):
```

Comment:
this could be a good time to uncomment this or add `test_update_volume_with_target_box_no_ref_values`, to test cases where there is unit specified for the target_box, probably the target_box array is a numpy array in that case.

Reply:
BTW, to make sure the previous error with pytest doesn't happen when initiating the simulation pass empty reference_values :
   sim_no_ref = Simulation(
             initial_state=benzene_system.hoomd_snapshot,
             forcefield=benzene_system.hoomd_forcefield,
            reference_values =dict()
        )

Reply:
Good idea, added this in the latest commit.